subt/cloudsim2osgar.py

```diff
@@ -260,6 +266,24 @@ def depth_rear(self, msg):
         rospy.loginfo_throttle(10, "depth_rear callback: {}".format(self.depth_rear_count))
         self.bus.publish('depth_rear', self.convert_depth(msg))
 
+    def convert_points(self, msg):
+        # accept only Velodyne VLC-16 (for the ver0)
+        assert msg.height == 16, msg.height
+        assert msg.width == 10000, msg.width
+        assert msg.point_step == 32, msg.point_step
+        assert msg.row_step == 320000, msg.row_step
+        ret = []
+        for i in range(msg.height * msg.width):
+            pt = struct.unpack_from('<ffff', msg.data, i * 32)
+            assert pt[3] == 0.0, pt
+            ret.append(pt[:3])
+        return ret
```

Comment:
My preference would be numpy array `msg.height x msg.width`. Could we use `np.frombuffer`?

What is stored inside of the four floats?

Reply:
I am not sure if our numpy msgpack types support float32bit, we will see. These are relative coordinates (x, y, z) and maybe reflectivity, which seems to be in Virtual always 0.0

Reply:
p.s. it looks like we support majority of `numpy` types ... but I would still drop the extra unused 0

Reply:
:+1: for dropping the unused 0 - otherwise I'd keep it as is (float32, meters) and used compression when saving it to the log